docs/pages/guides/keyboard-handling.mdx

```diff
@@ -0,0 +1,262 @@
+Keyboard handling is crucial for creating an excellent user experience in your Expo app. This guide covers common keyboard interactions and how to manage them effectively.
+
+React Native provides several keyboard-related APIs, with [`Keyboard`](https://reactnative.dev/docs/keyboard) and [`KeyboardAvoidingView`](https://reactnative.dev/docs/keyboardavoidingview) being the most frequently used. For more complex or custom keyboard interactions, you might consider using [Keyboard Controller](https://kirillzyusko.github.io/react-native-keyboard-controller), which is a library that offers advanced keyboard handling capabilities.
+
+## The basics
+
+### Keyboard events
+
+The `Keyboard` module from React Native allows you to listen for native events, react to them, and make changes to the keyboard, such as dismissing it.
+
+To listen for keyboard events, use the `Keyboard.addListener` method. This method accepts an event name and a callback function as arguments. The callback function is called with the event data when the keyboard is shown or hidden.
+
+```tsx HomeScreen.tsx
+import { useEffect } from 'react';
+import { Keyboard } from 'react-native';
+
+export default function HomeScreen() {
+  useEffect(() => {
+    const showSubscription = Keyboard.addListener('keyboardDidShow', handleKeyboardShow);
+
+    return () => {
+      showSubscription.remove();
+    };
+  }, []);
+
+  const handleKeyboardShow = event => {
+    // Handle keyboard show event here
+  };
+}
+```
+
+To dismiss the keyboard, use the `dismiss` method.
+
+```tsx DismissKeyboardButton.tsx
+import { Keyboard, Button } from 'react-native';
+
+export default function DismissKeyboardButton() {
+  return <Button title="Dismiss keyboard" onPress={Keyboard.dismiss} />;
+}
+```
+
+### Keyboard avoiding view
+
+The `KeyboardAvoidingView` is a component that automatically adjusts its height, position, or bottom padding based on the keyboard height to remain visible while the keyboard is displayed.
+
+```tsx HomeScreen.tsx
+import { KeyboardAvoidingView, TextInput } from 'react-native';
+
+export default function HomeScreen() {
+  return (
+    <KeyboardAvoidingView behavior={Platform.OS === 'ios' ? 'padding' : undefined} style={{ flex: 1 }}>
+      <TextInput placeholder="Type here..." />
+    </KeyboardAvoidingView>;
+  );
+}
+```
+
+In the above example, the height of the `KeyboardAvoidingView` automatically adjusts based on the device's keyboard height, which ensures that the input is always visible.
+
+When using a Bottom Tab navigator on Android, you might notice that focusing on an input field causes the bottom tabs to be pushed above the keyboard. To address this issue, add the `softwareKeyboardLayoutMode` property to your Android configuration in [app confg](/workflow/configuration/) and set it to `pan`.
+
+```json app.json
+"expo" {
+  "android": {
+    "softwareKeyboardLayoutMode": "pan"
+  }
+}
+```
+
+After adding this property, restart your app so that the changes take effect.
```

Comment:
This requires a development build and an actual rebuild of your app to take effect, right?

I was thinking we could also mention that `tabBarHideOnKeyboard` is an option that would still work with `adjustResize`, but might not look quite as smooth. https://reactnavigation.org/docs/bottom-tab-navigator/#tabbarhideonkeyboard

Reply:
It requires restarting the server; I need to make that clear. But this will work on Expo Go (no need for dev build)

Wow, I didn't know about `tabBarHideOnKeyboard`. We could mention it as an alternative, maybe.